lib/change_health/response/claim/report/report_835_service_line.rb

```diff
@@ -8,6 +8,46 @@ class Report835ServiceLine < Hashie::Trash
         property :line_item_charge_amount, required: false
         property :line_item_provider_payment_amount, required: false
         property :service_adjustments, required: false
+
+        def create_group_adjustments(service_adjustments)
+          adjustment_array = service_adjustments[:adjustments].map do |key, value|
+            {
+              adjustmentReasonCode: key,
+              adjustmentAmount: value
+            }
+          end
+          {
+            adjustmentDetails: adjustment_array,
+            adjustmentGroupCode: service_adjustments[:claim_adjustment_group_code]
+          }
+        end
+
+        def create_remark_code_adjustments(remark_codes_array)
+          adjustment_array = remark_codes_array.map do |_key, value|
+            {
+              adjustmentReasonCode: value,
+              adjustmentAmount: ""
+            }
+          end
+          {
+            adjustmentDetails: adjustment_array,
+            adjustmentGroupCode: ""
+          }
+        end
+
+        def create_adjustment_detail_array
+          adjustment_details = []
+          all_service_adjustments = self.service_adjustments
+          all_service_adjustments.each do |service_adjustments|
+            adjustment_details << create_group_adjustments(service_adjustments)
+          end
+
+          health_care_check_remark_codes = self[:health_care_check_remark_codes]
+          health_care_check_remark_codes.each do |remark_codes|
```

Comment:
Same nitpick as above.

Reply:
I changed it for the first one, but a second map just makes it nested when I push it into the same array. I could do a flatten, but just seems like extra code? I don't know if one is more efficient than another, but I like the idea of fewer lines

Reply:
You have to remove the `adjustment_details <<` I believe.

Reply:
Op my bad, I see what's happening, no map won't work here I guess. Sorry!